docs/SPEC.md

```diff
@@ -2793,6 +2915,49 @@ A query specification defines what data and operations to perform.
 The execution model reserves the right to perform those operations as efficiently as possible.
 The execution model may rewrite the query in anyway it sees fit while maintaining correctness.
 
+## Handling data errors
+
+Data errors are errors that occur when processing data from streams.
+When a data error occurs there are a few methods for handling the errors.
+
+First, there is the `fail` function that terminates the program.
+This can be used whenever an error occurs that is not recoverable in any form.
+
+Second, there is the `errors` function to provides direct access to the errors.
+Using the `errors` function any behavior Flux can describe is a possible method for handling errors.
+
+There is an option `errorHandler` that defines the default behavior for all data errors.
+The option value is a transformation function that consumes errors produced by other transformations.
+
+The default behavior is to fail when an error occurs.
+
+    option errorHandler = (tables=<-) =>
```

Comment:
or, perhaps this solves my problem from above about what `errors`  and `stats` should do.

Reply:
Really like this 👍